### PR TITLE
flake8 everything

### LINKED ISSUE #5880: flake8 everything
I mean everything.

### PR DESCRIPTION
**Pull request by mattpap**
fixes #5880

### CODE REVIEW

**Comment by bryevdv** [comment]
@mattpap I went ahead and merged master to resolve the conflict after the last PR. There were a couple of new flake8 error introduced by the merge as well. If you agree with my updates please merge.

File: tests/examples/test_examples.py [hunk 1]
```diff
@@ -182,6 +182,7 @@ def _assert_snapshot(example, url, example_type):
     success = result['success']
     timeout = result['timeout']
     errors = result['errors']
+    resources = result['resources']
 
     no_errors = len(errors) == 0
     no_resources = len(resources) == 0
```
Comment (bryevdv) [inline tests/examples/test_examples.py]: flake8 was complaining `resources` was undefined. I assume adding the line above was the intended solution.

File: tests/examples/examples_report_plugin.py [hunk 2]
```diff
@@ -1,13 +1,12 @@
 from __future__ import absolute_import, print_function
 
+import io
 import os
+from os.path import abspath, dirname, exists, expanduser, expandvars, join, pardir
```
Comment (bryevdv) [inline tests/examples/examples_report_plugin.py]: almost everywhere else we import individual funcs from `os.path` in fixing up this conflict I made that change as well.

File: tests/examples/test_examples.py (hunk 1 above)
Comment (mattpap) [inline tests/examples/test_examples.py]: Yes, because this PR removed it and the other (#5883) needed it back.

**Pull request merged**